alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportAcknowledgeChainedAlertAction.kt

```diff
@@ -0,0 +1,322 @@
+                                    )
+                                )
+                            )
+                        )
+                    } else {
+                        AcknowledgeHandler(client, actionListener, request).start(getWorkflowResponse.workflow!!)
+                    }
+                } catch (e: Exception) {
+                    log.error("Failed to acknowledge chained alerts from request $request", e)
+                    actionListener.onFailure(AlertingException.wrap(e))
+                }
+            }
+        }
+    }
+
+    inner class AcknowledgeHandler(
+        private val client: Client,
+        private val actionListener: ActionListener<AcknowledgeAlertResponse>,
+        private val request: AcknowledgeChainedAlertRequest,
+    ) {
+        val alerts = mutableMapOf<String, Alert>()
+
+        suspend fun start(workflow: Workflow) = findActiveAlerts(workflow)
+
+        private suspend fun findActiveAlerts(workflow: Workflow) {
+            try {
+                val queryBuilder = QueryBuilders.boolQuery()
+                    .must(
+                        QueryBuilders.wildcardQuery("workflow_id", request.workflowId)
+                    )
+                    .must(QueryBuilders.termsQuery("_id", request.alertIds))
+                if (workflow.inputs.isEmpty() || (workflow.inputs[0] is CompositeInput) == false) {
+                    actionListener.onFailure(
+                        OpenSearchStatusException("Workflow ${workflow.id} is invalid", RestStatus.INTERNAL_SERVER_ERROR)
+                    )
+                    return
+                }
+                val compositeInput = workflow.inputs[0] as CompositeInput
+                val workflowId = compositeInput.sequence.delegates[0].monitorId
+                val dataSources: DataSources = getDataSources(workflowId)
+                val searchRequest = SearchRequest()
+                    .indices(dataSources.alertsIndex)
+                    .routing(request.workflowId)
+                    .source(
+                        SearchSourceBuilder()
+                            .query(queryBuilder)
+                            .version(true)
+                            .seqNoAndPrimaryTerm(true)
+                            .size(request.alertIds.size)
+                    )
+
+                val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
+                onSearchResponse(searchResponse, workflow, dataSources)
+            } catch (t: Exception) {
+                log.error("Failed to acknowledge chained alert ${request.alertIds} for workflow ${request.workflowId}", t)
+                actionListener.onFailure(AlertingException.wrap(t))
+            }
+        }
+
+        private suspend fun getDataSources(monitorId: String): DataSources {
+            val getMonitorResponse: GetMonitorResponse =
+                transportGetMonitorAction.client.suspendUntil {
```

Comment:
This will have issues with RBAC. You can refer to this to replace this logic.

https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt#L574-L593

Reply:
replaced with getRequest from index